snuba/datasets/processors/spans_processor.py

```diff
@@ -0,0 +1,484 @@
+        user = _unicodify(tags.get("sentry:user", ""))
+        processed["user"] = user
+
+        common_span_fields["tags.key"] = ["environment", "release", "user"]
+        common_span_fields["tags.value"] = [span_environment, release, user]
+
+    @staticmethod
+    def _process_measurements(
+        processed: MutableMapping[str, Any],
+        event_dict: EventDict,
+    ) -> None:
+        """
+        Extracts measurements from the event_dict and writes them into the
+        measurements columns.
+        """
+        measurements = event_dict["data"].get("measurements")
+        if measurements is not None:
+            try:
+                (
+                    processed["measurements.key"],
+                    processed["measurements.value"],
+                ) = extract_nested(
+                    measurements,
+                    lambda value: float(value["value"])
+                    if (
+                        value is not None
+                        and isinstance(value.get("value"), numbers.Number)
+                    )
+                    else None,
+                )
+            except Exception:
+                logger.error(
+                    "Invalid measurements field.",
+                    extra={"measurements": measurements},
+                    exc_info=True,
+                )
+
+    @staticmethod
+    def _process_module_details(
+        processed: MutableMapping[str, Any],
+        event_dict: EventDict,
+    ) -> None:
+        """
+        TODO: For the top level span belonging to a transaction, we do not know how to fill these
+              values yet. For now lets just set them to their default values.
+        """
+        processed["module"] = ""
+        processed["action"] = ""
+        processed["domain"] = ""
+        processed["status"] = 0
+        processed["span_kind"] = ""
+
+    def _process_child_span_module_details(
+        self,
+        processed_span: MutableMapping[str, Any],
+        span_dict: SpanDict,
+    ) -> None:
+        """
+        TODO: This is a bit of a hack, but we need to extract some details from spans temporarily
+              until we get proper data from upstream. This code should be removed once we have
+              the proper data flowing through the pipeline.
+        """
+        op = span_dict.get("op", "")
+        description = span_dict.get("description", "")
+
+        # Op specific processing
+        if op != "db.redis" and op.startswith("db"):
+            parse_state = None
+            table = []
+
+            # This is wrong for so many reasons lol, but just something simple to pull some values out
+            raw_parsed = sqlparse.parse(description)[0]
+            processed_span["description"] = parse_query(
+                raw_parsed, "SAVEPOINT" in description.upper()
+            )
+            parsed = sqlparse.parse(processed_span["description"])[0]
```

Comment:
Why do we parse the query 3 times?

Reply:
See https://github.com/getsentry/snuba/pull/4102#discussion_r1180738350. I don't want to spend time optimizing this code right now. I will ask the product team to plug in whatever additional business logic they want to add in here in the future.